protocol-designer/src/pages/Designer/DeckSetup/SlotOverflowMenu.tsx

```diff
@@ -0,0 +1,207 @@
+    }
+    //  clear labware from slot
+    if (labwareOnSlot != null) {
+      dispatch(deleteContainer({ labwareId: labwareOnSlot.id }))
+    }
+    //  clear nested labware from slot
+    if (nestedLabwareOnSlot != null) {
+      dispatch(deleteContainer({ labwareId: nestedLabwareOnSlot.id }))
+    }
+  }
+
+  return createPortal(
+    <Overlay
+      data-testid="SlotOverflowMenu_Overlay"
+      onClick={(e: React.MouseEvent) => {
+        e.stopPropagation()
+        setShowMenuList(false)
+      }}
+    >
+      <Flex
+        whiteSpace="nowrap"
+        zIndex={10}
+        borderRadius={BORDERS.borderRadius8}
+        boxShadow="0px 1px 3px rgba(0, 0, 0, 0.2)"
+        position={POSITION_ABSOLUTE}
+        backgroundColor={COLORS.white}
+        //  todo(ja, 8/22/24): lol we need to fix these positions based off othe deck map & slot
+        right={ySlotPosition}
+        top={200 + xSlotPosition}
+        flexDirection={DIRECTION_COLUMN}
+        onClick={(e: React.MouseEvent) => {
+          e.preventDefault()
+          e.stopPropagation()
+        }}
+      >
+        <MenuButton
+          onClick={() => {
+            addEquipment(slot)
+            setShowMenuList(false)
+          }}
+        >
+          <StyledText desktopStyle="bodyDefaultRegular">
+            {t('add_hw')}
+          </StyledText>
+        </MenuButton>
+        <MenuButton
+          disabled={labwareOnSlot == null || isLabwareAnAdapter}
+          onClick={() => {
+            //  todo(ja, 8/22/24): wire this up
+            console.log('open nick name modal')
+            setShowMenuList(false)
+          }}
+        >
+          <StyledText desktopStyle="bodyDefaultRegular">
+            {t('rename_lab')}
+          </StyledText>
+        </MenuButton>
+        <MenuButton
+          disabled={labwareOnSlot == null || isLabwareTiprack}
+          onClick={() => {
+            if (labwareOnSlot != null && !isLabwareTiprack) {
+              //  todo(ja, 8/22/24): wire this up
+              console.log('open liquids')
+              setShowMenuList(false)
+            }
+          }}
+        >
+          <StyledText desktopStyle="bodyDefaultRegular">
+            {t('add_liquid')}
+          </StyledText>
+        </MenuButton>
+        <MenuButton
+          disabled={labwareOnSlot == null && !isLabwareAnAdapter}
+          onClick={() => {
+            if (labwareOnSlot != null && !isLabwareAnAdapter) {
+              dispatch(duplicateLabware(labwareOnSlot.id))
+              setShowMenuList(false)
+            }
+          }}
+        >
+          <StyledText desktopStyle="bodyDefaultRegular">
+            {t('duplicate')}
+          </StyledText>
+        </MenuButton>
+        <MenuButton
+          disabled={hasNoItems || hasTrashOnSlot}
+          onClick={() => {
+            if (!hasNoItems || !hasTrashOnSlot) {
+              handleClear()
+              setShowMenuList(false)
+            }
+          }}
+        >
```

Comment:
probably if can be removed because of `disabled`.

```suggestion
<MenuButton
  disabled={hasNoItems || hasTrashOnSlot}
  onClick={() => {
    handleClear();
    setShowMenuList(false);
  }}
>
```